l1-contracts/src/core/libraries/rollup/EpochProofLib.sol

```diff
@@ -119,6 +119,12 @@ library EpochProofLib {
     rollupStore.tips =
       rollupStore.tips.updateProvenBlockNumber(Math.max(rollupStore.tips.getProvenBlockNumber(), _args.end));
 
+    // Handle L2->L1 message processing
+    if (_args.args.outHash != bytes32(0)) {
+      // Insert L2->L1 messages into outbox for consumption.
+      rollupStore.config.outbox.insert(endEpoch, _args.args.outHash);
```

Comment:
Not something that I think we could have happen except with a catastrophic bug. But if we could have a partial epoch proof set a non-zero while the "full epoch" had been zero, it would not be overwritten here.